tgui/packages/tgui/interfaces/PreferencesMenu/LoadoutManager.tsx

```diff
@@ -0,0 +1,406 @@
+    loadout_tabs[0].name,
+  );
+
+  return (
+    <Stack vertical fill>
+      <Stack.Item>
+        <Section
+          title="Loadout Categories"
+          align="center"
+          buttons={
+            <Input
+              width="200px"
+              onInput={(_, value) => setSearchLoadout(value)}
+              placeholder="Search for an item..."
+              value={searchLoadout}
+            />
+          }
+        >
+          <Tabs fluid align="center">
+            {loadout_tabs.map((curTab) => (
+              <Tabs.Tab
+                key={curTab.name}
+                selected={
+                  searchLoadout.length <= 1 && curTab.name === selectedTabName
+                }
+                onClick={() => {
+                  setSelectedTab(curTab.name);
+                  setSearchLoadout('');
+                }}
+              >
+                {curTab.name}
+              </Tabs.Tab>
+            ))}
+          </Tabs>
+        </Section>
+      </Stack.Item>
+      <Stack.Item>
+        <LoadoutTabs
+          loadout_tabs={loadout_tabs}
+          currentTab={selectedTabName}
+          currentSearch={searchLoadout}
+        />
+      </Stack.Item>
+    </Stack>
+  );
+};
+
+const ItemIcon = (props: { item: LoadoutItem }) => {
+  const { item } = props;
+  const icon_to_use = item.icon;
+  const icon_state_to_use = item.icon_state;
+
+  if (!icon_to_use || !icon_state_to_use) {
+    return (
+      <Icon
+        name="question"
+        size={8}
+        color="red"
+        style={{ transform: 'translateX(6px) translateY(6px)' }}
+      />
+    );
+  }
+
+  return (
+    <DmIcon
+      fallback={<Icon name="spinner" spin color="gray" />}
+      icon={icon_to_use}
+      icon_state={icon_state_to_use}
+      style={{ transform: 'scale(3) translateX(10px) translateY(12px)' }}
+    />
+  );
+};
+
+// Is an item showing buttons?
+const ShowsButtons = (item: LoadoutItem, active: boolean) => {
+  for (const button of item.buttons) {
+    if (button.only_when_selected && !active) {
+      continue;
+    }
+    return true;
+  }
+  return false;
+};
+
+const ItemDisplayButton = (props: {
+  item: LoadoutItem;
+  button: LoadoutButton;
+  active: boolean;
+}) => {
+  const { act } = useBackend<LoadoutItem>();
+  const { item, button, active } = props;
+
+  if (button.only_when_selected && !active) {
+    return null;
+  }
+
+  return (
+    <Button
+      icon={button.icon}
+      height="22px"
+      width="22px"
+      tooltip={button.tooltip}
+      tooltipPosition={'bottom-start'}
+      disabled={!button.act_key}
+      color="yellow"
+      style={{ zIndex: '2' }}
+      onClick={(e) => {
+        e.stopPropagation();
+        act('pass_to_loadout_item', {
+          path: item.path,
+          subaction: button.act_key,
+        });
+      }}
+    />
+  );
+};
+
+const ItemDisplayButtons = (props: { item: LoadoutItem; active: boolean }) => {
+  const { item, active } = props;
+  const buttons = item.buttons;
```

Comment:
```suggestion
  const { buttons } = item;
```
Prefer destructuring pls